Sources/armory/renderpath/RenderPathDeferred.hx

```diff
@@ -838,11 +846,85 @@ class RenderPathDeferred {
 				path.bindTarget("_main", "tex");
 				path.drawShader("shader_datas/copy_pass/copy_pass");
 
+				path.setTarget("refr");
+				path.bindTarget("tex", "tex");
+				path.drawShader("shader_datas/copy_pass/copy_pass");
+
 				setTargetMeshes();
 				path.drawMeshes("refraction");
 
+				// ---
+				// Deferred light
+				// ---
+				#if (!kha_opengl)
+				path.setDepthFrom("tex", "gbuffer1"); // Unbind depth so we can read it
+				#end
+				path.setTarget("tex");
+				path.bindTarget("_main", "gbufferD");
+				path.bindTarget("gbuffer0", "gbuffer0");
+				path.bindTarget("gbuffer1", "gbuffer1");
+				
+				#if rp_gbuffer2
+				{
+					path.bindTarget("gbuffer2", "gbuffer2");
+				}
+				#end
+
+				#if rp_gbuffer_emission
+				{
+					path.bindTarget("gbuffer_emission", "gbufferEmission");
+				}
+				#end
+
+				#if (rp_ssgi != "Off")
+				{
+					if (armory.data.Config.raw.rp_ssgi != false) {
+						path.bindTarget("singlea", "ssaotex");
+					}
+					else {
+						path.bindTarget("empty_white", "ssaotex");
+					}
+				}
+				#end
+				var voxelao_pass = false;
+				#if rp_voxelao
+				if (armory.data.Config.raw.rp_gi != false)
+				{
+					#if arm_config
+					voxelao_pass = true;
+					#end
+					path.bindTarget(voxels, "voxels");
+					#if arm_voxelgi_temporal
+					{
+						path.bindTarget(voxelsLast, "voxelsLast");
+					}
+					#end
+				}
+				#end
+
+				#if rp_shadowmap
+				{
+					#if arm_shadowmap_atlas
+					Inc.bindShadowMapAtlas();
+					#else
+					Inc.bindShadowMap();
+					#end
+				}
+				#end
+
+				#if rp_material_solid
+				path.drawShader("shader_datas/deferred_light_solid/deferred_light");
+				#elseif rp_material_mobile
+				path.drawShader("shader_datas/deferred_light_mobile/deferred_light");
+				#else
+				voxelao_pass ?
+					path.drawShader("shader_datas/deferred_light/deferred_light_VoxelAOvar") :
+					path.drawShader("shader_datas/deferred_light/deferred_light");
+				#end
+
```

Comment:
Do I understand this code correctly that you basically render the scene twice if there are refractive materials?